machineconfig/master/spec/config/storage/files/keepalived.sh L3Vzci9sb2NhbC9iaW4va2VlcGFsaXZlZC5zaAo=

```diff
@@ -0,0 +1,60 @@
+#!/usr/bin/env bash
+set -e
+
+function get_iface_in_vip_subnet() {
+    local vip
+    local net_cidr
+    local iface_cidrs
+
+    vip="$1"
+
+    iface_cidrs=$(ip addr show | grep -v "scope host" | grep -Po 'inet \K[\d.]+/[\d.]+' | xargs)
+    net_cidr=$(/usr/bin/env python - "$vip" "$iface_cidrs" << EOF
+import sys
+import socket
+import struct
+
+vip = sys.argv[1]
+iface_cidrs = sys.argv[2].split()
+vip_int = struct.unpack("!I", socket.inet_aton(vip))[0]
+
+for iface_cidr in iface_cidrs:
+    ip, prefix = iface_cidr.split('/')
+    ip_int = struct.unpack("!I", socket.inet_aton(ip))[0]
+    prefix_int = int(prefix)
+    mask = int('1' * prefix_int + '0' * (32 - prefix_int), 2)
+    subnet_ip_int_min = ip_int & mask
+    subnet_ip = socket.inet_ntoa(struct.pack("!I", subnet_ip_int_min))
+    subnet_ip_int_max = subnet_ip_int_min | int('1' * (32 - prefix_int), 2)
+    subnet_ip_max = socket.inet_ntoa(struct.pack("!I", subnet_ip_int_max))
+    sys.stderr.write('Is %s between %s and %s\n' % (vip, subnet_ip, subnet_ip_max))
+    if subnet_ip_int_min < vip_int < subnet_ip_int_max:
+        subnet_ip = socket.inet_ntoa(struct.pack("!I", subnet_ip_int_min))
+        print('%s/%s' % (subnet_ip, prefix))
+        sys.exit(0)
+sys.exit(1)
+EOF
+)
+    ip -o addr show to "$net_cidr" | awk '{print $2}'
+}
+
+mkdir --parents /etc/keepalived
+
+KEEPALIVED_IMAGE=registry.access.redhat.com/rhosp14/openstack-keepalived:14.0
+if ! podman inspect "$KEEPALIVED_IMAGE" &>/dev/null; then
+    echo "Pulling release image..."
+    podman pull "$KEEPALIVED_IMAGE"
+fi
+
+INTERFACE="$(get_iface_in_vip_subnet "$API_VIP")" "API_VIP=$API_VIP" envsubst < /etc/keepalived/keepalived.conf.template > /etc/keepalived/keepalived.conf
```

Comment:
```suggestion
env INTERFACE="$(get_iface_in_vip_subnet "$API_VIP")" "API_VIP=$API_VIP" envsubst < /etc/keepalived/keepalived.conf.template > /etc/keepalived/keepalived.conf
```

Reply:
Here it is not necessary because we explicitly set the env vars that envsubst should substitute.